packages/netstring/reader.js

```diff
@@ -2,67 +2,113 @@
 /// <reference types="ses"/>
 
 const COLON = ':'.charCodeAt(0);
-
-const decoder = new TextDecoder();
+const COMMA = ','.charCodeAt(0);
+const ZERO = '0'.charCodeAt(0);
+const NINE = '9'.charCodeAt(0);
 
 /**
  * @param {Iterable<Uint8Array> | AsyncIterable<Uint8Array>} input
  * @param {Object} [opts]
  * @param {string} [opts.name]
- * @param {number} [opts.capacity]
  */
-async function* makeNetstringIterator(
-  input,
-  { name = '<unknown>', capacity = 1024 } = {},
-) {
-  let length = 0;
-  let buffer = new Uint8Array(capacity);
+async function* makeNetstringIterator(input, { name = '<unknown>' } = {}) {
```

Comment:
I’ve heard that generators are slow and that state machines are the way to go for speed. I used a generator in hopes that a the readability would be more important than speed. This appears to be the time to revisit that choice. But, I like testing the fast one in comparison to the slow-but-readable one for correctness, especially in a fuzzer.

Reply:
Yeah since we already have a state machine inside the generator, you're right going all the way might be better.

Reply:
Decided to not do as consuming the input without a `for await of` loop makes things more complicated.